include/receiver_cadmium.hpp

```diff
@@ -27,44 +36,66 @@
 using namespace cadmium;
 using namespace std;
 
-//Port definition
+/** 
+* Structure that holds input and output messages.
+*/
 struct receiver_defs {
     struct out : public out_port<Message_t> {
     };
     struct in : public in_port<Message_t> {
     };
 };
-   
+
+/** 
+* The Receiver class receives message and sends out acknowledge.
+*/
 template<typename TIME>
 class Receiver {
-    using defs = receiver_defs; // putting definitions in context
+    // putting definitions in context
+    using defs = receiver_defs;
     public:
-        //Parameters to be overwriten when instantiating the atomic model
-        TIME PREPARATION_TIME;
-        // default constructor
+        TIME PREPARATION_TIME;   /**< Constant that holds the time delay from input to output. */ //!<Time delay constant.
+		
+        /** 
+	* Constructor for Receiver class.
+	* Initializes the delay constant and state structure.
+	*/
         Receiver() noexcept {
             PREPARATION_TIME  = TIME("00:00:10");
             state.ack_num     = 0;
             state.sending     = false;
         }
             
-        // state definition
+        /**
+	* Structure that holds acknowledge number and receiver state.
+	*/
         struct state_type {
-            int ack_num;
-            bool sending;
+            int ack_num;   /**< Alternating bit retrieved from the message and sent as acknowledge. */ //!< Acknowledge Number.
+            bool sending;  /**< State of the receiver: true - sending, false - passive. */ //!< State of the receiver.
         }; 
-        state_type state;
+        state_type state; 
 		
         // ports definition
         using input_ports = std::tuple<typename defs::in>;
         using output_ports = std::tuple<typename defs::out>;
 
-        // internal transition
+        /**
+	* Function that performs the internal transition.
+        * It turns off the receiver sending state.
+	*/
         void internal_transition() {
             state.sending = false; 
         }
 
-        // external transition
+        /**
+	* Function that performs external transition.
+	* Retrieves the messages: if the number of messages
+	* is more than 1, it asserts that only one message is
+        * expected per time unit. It then sets the acknowledge
+	* to the message value and state to sending.
```

Comment:
I guess you can include that the state of receiver will be active/true.

Reply:
Done